beater/waitready.go

```diff
@@ -79,10 +81,17 @@ func waitReady(
 type waitReadyRoundTripper struct {
 	*http.Transport
 	ready <-chan struct{}
+	drain <-chan struct{}
 }
 
 func (c *waitReadyRoundTripper) RoundTrip(r *http.Request) (*http.Response, error) {
 	select {
+	case <-c.drain:
+		return &http.Response{
+			Request:    r,
+			StatusCode: http.StatusPreconditionFailed,
+			Body:       io.NopCloser(new(bytes.Buffer)),
```

Comment:
```suggestion
			Body:       io.NopCloser(strings.NewReader("server shutting down")),
```

might be a bit more understandable in logs?